### ISSUE TITLE
Updating the connect NPM dependency to the current version

### ISSUE THREAD

**Issue by Kriegslustig** [meteor/meteor#4878]
[Connect](https://github.com/senchalabs/connect) has gone through a lot of changed a lot in the last [two years](https://github.com/senchalabs/connect/releases/tag/2.9.0). I'd say, for the better. [WebApp](https://github.com/meteor/meteor/tree/devel/packages/webapp) still depends on version [2.9.0](https://github.com/meteor/meteor/blob/devel/packages/webapp/package.js#L6). The update is long overdue.

The core Connect Package is a lot slimmer and the middleware that is used by Meteor has probably also been improved. It's split up into multiple packages now.

PS: I'd love to do it myself :) Wanted to check first if there's a big reason for it being so outdated.

**Comment by stubailo** [comment]
Thanks for the feature request! We welcome discussions about how to make Meteor better. If you haven't yet, check out our wiki page about [feature requests on GitHub](https://github.com/meteor/meteor/wiki/Contributing-to-Meteor#feature-requests).

I personally think this would be a really awesome thing to do, and we should do a better job of keeping our dependencies up to date.

Some concerns/questions:

1. Should we just switch to Express? It seems like a lot of middlewares are not build for Express, not Connect. Also, it might make it super easy to handle HTTP requests with Meteor if we expose the Express API in a reasonable way.
2. What are the changes in Connect since 2.9.0? We should see if there is something super breaking that will make middlewares written for WebApp not work anymore.

**Comment by Kriegslustig** [comment]
Thanks for your quick response :)
I'll write up some pros and cons to integrating Express into meteor.

**Comment by Kriegslustig** [comment]
[Here](https://gist.github.com/Kriegslustig/c6e2355be4ffbb324dc3)'s the list I made.
The points listed are (_mostly_) facts. Here's my take on the question:

Connect provides exactly what we need. An discrete, extensible component which handles HTTP requests. Nothing more and nothing less. Express is a nice point for people to start off. You don't have to go through the hassle of evaluating what middlewares/components you need and how they hold up qualitatively. Meteor's use-case is very different. We can easily put the time into evaluating middlewares for it. I believe, the fact that Sencha still actively maintains `connect` supports this argument.

Another point I raised above, is that of routing. Express comes with a server-side solution. When writing the Meteor core router, the following decision would have to be made: Should we lock our-selfs in to Express's syntax, or should we bake our own solution and leave Express's routing solution lying around. I know that this argument isn't all to strong, but I think it adds to the other two.

There are more middlewares written for Express that for Connect. But relative to their popularity, Connect seems to be better of. Newer might be exclusively written for Express but porting these to Connect is mostly pretty easy. Sometimes it can even be done by simply adding some other middleware (like [qs](https://www.npmjs.com/package/qs)).

**TL;DR; I would update connect, integrate more of it's middlewares and provide a nicer API to Connect.**

**Comment by hwillson** [comment]
Just a quick update on this one - WebApp has since been updated to use connect version 2.30.2, the last 2.x version. I'm assuming version 3 isn't being used yet due to breaking changes, but I haven't verified this yet myself.

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Updating%20the%20connect%20NPM%20dependency%20to%20the%20current%20version&body=Migrated%20from:%20meteor/meteor%234878) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
